ios/cocoapods_best_practices.md

```diff
@@ -0,0 +1,61 @@
+# CocoaPods Best Practices
+
+CocoaPods is a vital tool within our iOS ecosystem, and one that is used on almost every project built at Intrepid. For that reason it's important that we also have a set of guidelines and best practices in place for working with CocoaPods to ensure relative conformity across various projects.
+
+## Structuring a Podfile
+
+Podfiles should have as much structure to them as possible. CocoaPods makes things very easy and very straightforward to get started, and you aren't required to be restrictive in the versions you use, but that can easily cause chaos on a project when a new developer rolls on or a new version of the pod is released without being fully tested. For those reasons it's strongly encouraged to be as exact as possible. 
+
+#### Bad Podfile example
+```ruby
+target 'Project' do
+    pod 'RxCocoa'
+    pod 'RxSwift'
+    pod 'Analytics'
+    pod 'Crashlytics'
+    pod 'OHHTTPStubs/Swift'
+    pod 'Firebase/Core'
+    pod 'Alamofire'
+    pod 'Fabric', '~> 1.7'
+    pod 'IGListKit', '~> 3.1'
+end
+```
+This Podfile is very quick to set up, and very easy to use. That ease comes with a tradeoff though where you aren't guaranteeing the version that will be installed when `pod install` is run. Sure, for Fabric and IGListKit you are giving a version, but they will bring in any version up to `2.0` and `4.0` respectively. Granted the maintainers should be adhering to semver style and shouldn't be introducing any major breaking changes in those versions, but in the event of that happening your project could be broken unexpectedly.
```

Comment:
> That ease comes with a tradeoff though where you aren't guaranteeing the version that will be installed when `pod install` is run.

The versioning is well defined, and there may be a case where you want to install any version up to and including that.  Your pod will not automatically update the version on `pod install` if you have a `Podfile.lock` in the repository.  It only updates on `pod upgrade`

https://guides.cocoapods.org/using/pod-install-vs-update.html

Reply:
Oh interesting. I didn't know about the differences between those two. This only adds an extra guarantee against a run of `pod upgrade` and is unnecessary for `pod install` then. Unless the project isn't checking in pods and doesn't check in the `podfile.lock` file.